main_test.go

```diff
@@ -19,35 +21,38 @@ func TestFetchUnprocessedRecords(t *testing.T) {
 	// TODO: Use actual trigger to generate this?
 	events := []*eventqueue.Event{
 		{
-			ExternalID: []byte("fefc72b4-d8df-4039-9fb9-bfcb18066a2b"),
-			TableName:  "users",
-			Statement:  "UPDATE",
-			Data:       []byte(`{ "email": "j@blendle.com" }`),
-			Processed:  true,
+			ExternalID:    []byte("fefc72b4-d8df-4039-9fb9-bfcb18066a2b"),
+			TableName:     "users",
+			Statement:     "UPDATE",
+			State:         []byte(`{ "email": "j@blendle.com" }`),
+			ChangedFields: []string{"email"},
+			Processed:     true,
 		},
 		{
-			ExternalID: []byte("fefc72b4-d8df-4039-9fb9-bfcb18066a2b"),
-			TableName:  "users",
-			Statement:  "UPDATE",
-			Data:       []byte(`{ "email": "jurre@blendle.com" }`),
+			ExternalID:    []byte("fefc72b4-d8df-4039-9fb9-bfcb18066a2b"),
+			TableName:     "users",
+			Statement:     "UPDATE",
+			State:         []byte(`{ "email": "jurre@blendle.com" }`),
+			ChangedFields: []string{"email"},
 		},
 		{
 			ExternalID: []byte("fefc72b4-d8df-4039-9fb9-bfcb18066a2b"),
 			TableName:  "users",
 			Statement:  "UPDATE",
-			Data:       []byte(`{ "email": "jurres@blendle.com" }`),
+			State:      []byte(`{ "email": "jurres@blendle.com" }`),
 		},
 		{
 			ExternalID: nil,
 			TableName:  "users",
 			Statement:  "CREATE",
-			Data:       []byte(`{ "email": "bart@simpsons.com" }`),
+			State:      []byte(`{ "email": "bart@simpsons.com" }`),
```

Comment:
is `ChangedFields` nil in this test?

Reply:
In the CREATE statement events we don't care about the changed_fields. It will be nil practically.